judge/forms.py

```diff
@@ -95,6 +97,28 @@ def __init__(self, *args, **kwargs):
         self.fields['user_script'].widget = AceWidget(mode='javascript', theme=user.profile.resolved_ace_theme)
 
 
+class EmailChangeForm(Form):
+    password = CharField(widget=forms.PasswordInput())
```

Comment:
Obviously, all this explodes for social auth. Do we have a notice about this? What if a user has social auth but also a password?

Reply:
We don't right now, but we also don't have anything set up for password change. In a future PR, we should just hide these endpoints if the user has no password set and say something about using social auth.

Right now, I'm of the opinion that if a user is using social auth but has a password set, then they should be treated as a normal user and be allowed to change their emails.

Reply:
What happens when a user with social auth changes their email? Does their social auth still work? If so I think it's probably fine.

Reply:
Yes it should be fine. I don't believe social auth uses this email field after initially populating it.